services/export/shared/transactionsExport.js

```diff
@@ -0,0 +1,462 @@
+		chainID = transaction && transaction.params.sendingChainID
+			? transaction.params.sendingChainID
+			: chainID;
+	}
+
+	const date = dateFromTimestamp(tx.block.timestamp);
+	const time = timeFromTimestamp(tx.block.timestamp);
+	const amount = normalizeTransactionAmount(address, tx);
+	const fee = normalizeTransactionFee(address, tx);
+	const amountTokenID = tx.params.tokenID;
+	const senderAddress = tx.sender.address;
+	const recipientAddress = tx.params.recipientAddress || null;
+	const recipientPublicKey = tx.meta && tx.meta.recipient && tx.meta.recipient.publicKey || null;
+	const blockHeight = tx.block.height;
+	const note = tx.params.data || null;
+	const transactionID = tx.id;
+	const sendingChainID = chainID;
+	const receivingChainID = resolveReceivingChainID(tx, chainID);
+
+	return {
+		date,
+		time,
+		amount,
+		fee,
+		amountTokenID,
+		moduleCommand,
+		senderAddress,
+		recipientAddress,
+		senderPublicKey,
+		recipientPublicKey,
+		blockHeight,
+		note,
+		transactionID,
+		sendingChainID,
+		receivingChainID,
+	};
+};
+
+const exportTransactions = async (job) => {
+	const { params } = job.data;
+
+	const allTransactions = [];
+
+	const interval = await standardizeIntervalFromParams(params);
+	const [from, to] = interval.split(':');
+	const range = moment.range(moment(from, DATE_FORMAT), moment(to, DATE_FORMAT));
+	const arrayOfDates = (Array.from(range.by('day'))).map(d => d.format(DATE_FORMAT));
+
+	for (let i = 0; i < arrayOfDates.length; i++) {
+		/* eslint-disable no-await-in-loop */
+		const day = arrayOfDates[i];
+		const partialFilename = await getPartialFilenameFromParams(params, day);
+		if (await partials.exists(partialFilename)) {
+			const transactions = JSON.parse(await partials.read(partialFilename));
+			allTransactions.push(...transactions);
+		} else if (await noTransactionsCache.get(partialFilename) !== true) {
+			const fromTimestampPast = moment(day, DATE_FORMAT).startOf('day').unix();
+			const toTimestampPast = moment(day, DATE_FORMAT).endOf('day').unix();
+			const timestamp = `${fromTimestampPast}:${toTimestampPast}`;
+			const { data: transactions } = await requestAll(
+				getTransactionsInAsc, { ...params, timestamp },
+				MAX_NUM_TRANSACTIONS,
+			);
+			allTransactions.push(...transactions);
+
+			const incomingCrossChainTransferTxs = await getCrossChainTransferTransactionInfo({
+				...params,
+				timestamp,
+			});
+
+			if (incomingCrossChainTransferTxs.length) {
+				allTransactions.push(...incomingCrossChainTransferTxs);
+			}
+
+			if (day !== getToday()) {
+				if (transactions.length) {
+					partials.write(partialFilename, JSON.stringify(transactions));
+				} else {
+					// Flag to prevent unnecessary calls to core/storage space usage on the file cache
+					const RETENTION_PERIOD = getDaysInMilliseconds(config.cache.partials.retentionInDays);
+					await noTransactionsCache.set(partialFilename, true, RETENTION_PERIOD);
```

Comment:
```suggestion
					// Flag to prevent unnecessary calls to indexer/storage space usage on the file cache
					const RETENTION_PERIOD_MS = getDaysInMilliseconds(config.cache.partials.retentionInDays);
					await noTransactionsCache.set(partialFilename, true, RETENTION_PERIOD_MS);
```